source/material_model/viscoelastic_plastic.cc

```diff
@@ -0,0 +1,796 @@
+                                                                       n_fields,
+                                                                       "Viscosities");
+          thermal_conductivities = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Thermal conductivities"))),
+                                                                           n_fields,
+                                                                           "Thermal conductivities");
+          thermal_expansivities = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Thermal expansivities"))),
+                                                                          n_fields,
+                                                                          "Thermal expansivities");
+          specific_heats = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Specific heats"))),
+                                                                   n_fields,
+                                                                   "Specific heats");
+          elastic_shear_moduli = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Elastic shear moduli"))),
+                                                                         n_fields,
+                                                                         "Elastic shear moduli");
+
+          if (prm.get ("Use fixed elastic time step") == "true")
+            use_fixed_elastic_time_step = true;
+          else if (prm.get ("Use fixed elastic time step") == "false")
+            use_fixed_elastic_time_step = false;
+          else
+            AssertThrow(false, ExcMessage("'Use fixed elastic time step' must be set to 'true' or 'false'"));
+
+          use_stress_averaging = prm.get_bool ("Use stress averaging");
+          if (use_stress_averaging)
+            AssertThrow(use_fixed_elastic_time_step == true,
+                        ExcMessage("Stress averaging can only be used if 'Use fixed elastic time step' is set to true'"));
+
+          fixed_elastic_time_step = prm.get_double ("Fixed elastic time step");
+          AssertThrow(fixed_elastic_time_step > 0,
+                      ExcMessage("The fixed elastic time step must be greater than zero"));
+
+          if (this->convert_output_to_years())
+            fixed_elastic_time_step *= year_in_seconds;
+
+          AssertThrow(this->get_parameters().enable_elasticity == true,
+                      ExcMessage ("Material model Viscoelastic only works if 'Enable elasticity' is set to true"));
+
+          // Check whether the compositional fields representing the viscoelastic
+          // stress tensor are both named correctly and listed in the right order.
+          if (dim == 2)
```

Comment:
You can remove this if, because these three asserts are the same for 2d and 3d. Also remove them from the dim==3 case and turn that into an if.

Reply:
The third assert for the 2D case (`"stress_xy"`) is unfortunately different between the 2D and 3D cases. I could take the first two cases (`stress_xx` and `stress_yy`) out of the if statement, but I would still need an if statement for the third component of the 2D case.